sycl/include/sycl/queue.hpp

```diff
@@ -1068,6 +1083,22 @@ class __SYCL_EXPORT queue {
   /// \return the backend associated with this queue.
   backend get_backend() const noexcept;
 
+public:
+  /// Places the queue into command_graph recording mode.
+  /// Returns true if the queue was not already in recording mode.
+  bool begin_recording(
+      sycl::ext::oneapi::experimental::command_graph<
```

Comment:
we might be able to drop the `sycl::` part of the namespace from the start of this type (and others you've added)

Reply:
Done!